### ISSUE TITLE
Build deletion fails due to buildartifacttype schema validation error

### ISSUE THREAD

**Issue by sblair-metrostar** [conda-incubator/conda-store#420]
On two ~1 month old Nebari installations, I've observed that conda-store build deletions triggered through the UI fail with the following error on the worker pod:

![image](https://user-images.githubusercontent.com/70233904/212155492-bfc3455a-637c-4534-a2a6-051332fc1e3a.png)

Inspecting the postgres database confirms that CONTAINER_REGISTRY value is missing from the type definition. It appears it should have been added based on the alembic migration [here](https://github.com/Quansight/conda-store/blob/main/conda-store-server/conda_store_server/alembic/versions/5ad723de2abd_adding_container_registry_value_to_enum.py) and the version reported in the alembic_version table:

![image](https://user-images.githubusercontent.com/70233904/212156676-98dfdf29-81e9-4a3a-be5c-4d6d11f44584.png)

I'm not sure where the gap might be. If I manually add the value to the type definition, deletions appear to work as far as I can tell:

`ALTER TYPE buildartifacttype ADD VALUE 'CONTAINER_REGISTRY' AFTER 'DOCKER_MANIFEST'`

![image](https://user-images.githubusercontent.com/70233904/212157266-6926b8e9-b465-4f0b-8db5-61a59e993ffe.png)

**Comment by kcpevey** [comment]
We are still seeing this on the backend, though on the admin interface, the builds are showing up as deleted:

![image](https://github.com/conda-incubator/conda-store/assets/15659110/1fdc6ff8-4ed6-4ef8-91eb-3d5d83cfd9b7)

**Comment by iameskild** [comment]
@pavithraes @trallard is there bandwidth to include a fix for this in the upcoming conda-store release? This blocks users from deleting envs/builds.

**Comment by kcpevey** [comment]
We confirmed that the fix about fixes the error on the delete process. Now we see the delete process succeeding. 

```
[2023-09-05 21:11:10,476: INFO/MainProcess] Task task_delete_build[d12572cf-694d-4341-86ab-34a21dac779f] received
[2023-09-05 21:11:10,482: WARNING/ForkPoolWorker-2] [CondaStoreWorker] deleting artifacts for build=89
[2023-09-05 21:11:13,712: INFO/ForkPoolWorker-2] Task task_delete_build[d12572cf-694d-4341-86ab-34a21dac779f] succeeded in 3.235292388126254s: None
```

However, our storage size is still the same. I am going to open a separate issue for that.

**Comment by nkaretnikov** [comment]
Metadata update (size) and misc Python errors related to deletion were fixed in https://github.com/conda-incubator/conda-store/pull/572.
Postgres migration issue was addressed in https://github.com/conda-incubator/conda-store/pull/574.

I've also verified on a8d352443795a6660e0344c93fd43de24c171ebe, which is on top of 432eaa11ef349fdb848215113caea22b09a23a32, which has both fixes, that I can:
- build the default env in the filesystem namespace
- add another env in user namespace with the same spec (but different name)
- build one env, check its artifacts and blobs in minio 
- remove the same env, data in minio is deleted
- same for another one in user namespace
- then build both envs and test each other separately (to test potential issues with pattern-matching), all data gone in the end.

There's another issue that we may potentially want to address, in `delete_build_artifact` CONTAINER_REGISTRY is skipped. I'll open a new issue since that's separate.

I think this issue can be closed.
